core/http/client/src/main/java/org/eclipse/rdf4j/http/client/SesameClientImpl.java

```diff
@@ -107,25 +125,26 @@ public synchronized SesameSession createSesameSession(String serverURL) {
 	 *-----------------*/
 
 	@Override
-	public synchronized void shutDown() {
-		if (executor != null) {
+	public void shutDown() {
+		try {
 			executor.shutdown();
-			executor = null;
 		}
-		if (dependentClient != null) {
-			HttpClientUtils.closeQuietly(dependentClient);
+		finally {
+			CloseableHttpClient toCloseDependentClient = dependentClient;
 			dependentClient = null;
+			if (toCloseDependentClient != null) {
+				HttpClientUtils.closeQuietly(toCloseDependentClient);
+			}
 		}
 	}
 
 	/**
-	 * (re)initializes the connection manager and HttpClient (if not already done), for example after a
-	 * shutdown has been invoked earlier. Invoking this method multiple times will have no effect.
+	 * No-op
+	 * 
+	 * @deprecated Create a new instance instead of trying to reactivate an old instance.
 	 */
-	public synchronized void initialize() {
-		if (executor == null) {
-			executor = Executors.newCachedThreadPool();
-		}
+	@Deprecated
+	public void initialize() {
```

Comment:
There was no instance of this method being called, according to Eclipse Call Heirarchy. The code that was there is easy enough to reinstate if necessary. If it turns out that people do need to reinitialise these, instead of replacing them with new copies, then I will need to change some of the other parts of the code also. 

I am not happy with the pull request yet, but it is great to get comments on it so far. In particular, the sample code referenced the .evaluate method that I haven't inspected closely yet.